tidy3d/components/data/dataset.py

```diff
@@ -942,8 +997,17 @@ def interp(
             y-coordinates of sampling points.
         z : Union[float, ArrayLike]
             z-coordinates of sampling points.
-        fill_value : float = 0
-            Value to use when filling points without interpolated values.
+        fill_value : Union[float, Literal["extrapolate"]] = "extrapolate"
+            Value to use when filling points without interpolated values. If ``extrapolate`` then
+            nearest values are used.
+        use_vtk : bool = False
+            Use vtk's interpolationfunctionality or Tidy3D's own implementation.
+        method: Literal["linear", "nearest"] = "linear"
+            Interpolation method to use.
+        max_samples_per_step : int = 1e4
```

Comment:
Some of these parameters are rather technical and used only in our own interpolation. Maybe @tylerflex has a better suggestion, but one possibility is to make these parameters kwargs here, and make the `interp_vtk` and `interp_py` functions public so that the `interp_py` docstring would have the details of these parameters